server/global/images_importation/imagesAnayzer.js

```diff
@@ -0,0 +1,129 @@
+import { queryPromise } from "../../db/database.js";
+import { removeExtension } from "../Files.js";
+import { addErrorTitle } from "../Logger.js";
+
+import {
+  normalizeDCI,
+  getInvalidNormalizedDci,
+} from "../molecules_importation/moleculesAnalyzer.js";
+
+const VALID_FORMATS = ["jpeg", "jpg", "svg", "png"];
+export const isFormatValid = (filename) =>
+  new RegExp(`\\.${VALID_FORMATS.join("|")}\\s*$`, "i").test(filename);
+
+/**
+ * Analyze images filenames
+ * @param {string[]} filenames The list of images filenames
+ * @returns {Promise<ImagesAnalyzerWarning[]>} The warnings list
+ */
+export function analyseImagesFilenames(filenames) {
+  return new Promise((resolve, reject) => {
+    const warnings = [];
+
+    const invalidFormats = filenames.filter((f) => !isFormatValid(f));
```

Comment:
I think we could also rely on the [MIME type](https://developer.mozilla.org/fr/docs/Web/HTTP/Basics_of_HTTP/MIME_Types) and not only on the file extension, as the former is more reliable.

Reply:
I did this initially, but in reality there are some non-standard MIME TYPE that we cannot handle, for example for CSV files the standard is `text/csv`, but also `application/vnd.ms-excel`, and surely others. It is therefore difficult for us to verify this, at the risk of refusing a valid file.
For our use, only admins will import the files, so we can trust them. In addition, they will receive an import manual explaining how to do it correctly.